### PR TITLE
adding basic sessions app

### PR DESCRIPTION
**Pull request by flossypurse**

### CODE REVIEW

File: sessions/workflow_dacx.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	if err != nil {
+		return err
+	}
+	uploadParam := FileActivityParam(processResult)
+	err = workflow.ExecuteActivity(sessionCtx, a.UploadFile, uploadParam).Get(sessionCtx, nil)
+	return err
+}
+
+/*
+When the `CreateSession` API is called, the Task Queue name that is specified in the `ActivityOptions` is used (or in the `StartWorkflowOptions` if the Task Queue name is not specified in `ActivityOptions`), and a Session is created with one of the Workers polling that Task Queue.
```
Comment (djmagee) [inline sessions/workflow_dacx.go:67]:
```suggestion
When the `CreateSession` API is called, the Task Queue name that is specified in `ActivityOptions` (or in `StartWorkflowOptions` if the Task Queue name is not specified in `ActivityOptions`) is used, and a Session is created with one of the Workers polling that Task Queue.
```

File: sessions/workflow_dacx.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+The Session Context is cancelled if the Worker executing this Session dies or `CompleteSession()` is called.
+When using the returned Session Context to spawn Activity Executions, a `workflow.ErrSessionFailed` error may be returned if the Session framework detects that the Worker executing this Session has died.
```
Comment (djmagee) [inline sessions/workflow_dacx.go:70]:
```suggestion
When using the returned Session Context to spawn Activity Executions, a `workflow.ErrSessionFailed` error is returned if the Session framework detects that the Worker executing this Session has died.
```
Or are there situations in which the `workflow.ErrSessionFailed` error isn't returned?

File: sessions/workflow_dacx.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+The failure of Activity Executions won't affect the state of the Session, so you still need to handle the errors returned from your Activities and call `CompleteSession()` if necessary.
+
+`CreateSession()` will return an error if the context passed in already contains an open Session.
```
Comment (djmagee) [inline sessions/workflow_dacx.go:73]:
```suggestion
If the context passed in already contains an open Session, `CreateSession()` returns an error.
```

File: sessions/workflow_dacx.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+If all the Workers are currently busy and unable to handle a new Session, the framework will keep retrying until the `CreationTimeout` you specified in `SessionOptions` has passed before returning an error (check the **Concurrent Session Limitation** section for more details).
```
Comment (djmagee) [inline sessions/workflow_dacx.go:74]:
```suggestion
If all the Workers are currently busy and unable to handle a new Session, the framework keeps retrying until the `CreationTimeout` period you specified in `SessionOptions` has passed before returning an error.
(For more details, check the "Concurrent Session Limitation" section.)
```

File: sessions/workflow_dacx.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+`CompleteSession()` releases the resources reserved on the Worker, so it's important to call it as soon as you no longer need the Session.
+It will cancel the session context and therefore all the Activity Executions using that Session Context.
```
Comment (djmagee) [inline sessions/workflow_dacx.go:77]:
```suggestion
It cancels the session context and therefore all the Activity Executions using that Session Context.
```

File: sessions/workflow_dacx.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+It is safe to call `CompleteSession()` on a failed Session, meaning that you can call it from a `defer` function after the Session is successfully created.
+
+If the Worker goes down between Activities, any scheduled Activities meant for the Session Worker are canceled.
+If not, you will get a `workflow.ErrSessionFailed` error when the next call of `workflow.ExecuteActivity()` is made from that Workflow.
```
Comment (djmagee) [inline sessions/workflow_dacx.go:81]:
```suggestion
If not, you get a `workflow.ErrSessionFailed` error when the next call of `workflow.ExecuteActivity()` is made from that Workflow.
```
To what situation does "If not" refer? A failure of scheduled Activities to be canceled?

File: sessions/workflow_dacx.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+*/
+
+/* @dacx
+id: how-to-create-a-session-from-the-workflow-in-go
+title: How to create a Worker Session in Go
+label: Worker Session
+description: First, enable Sessions on the Worker via the Worker Options and then use the `CreateSession` API to create a Context object that can be passed to the calls to spawn Activity Executions.
```
Comment (djmagee) [inline sessions/workflow_dacx.go:88]:
```suggestion
description: Enable Sessions on the Worker via the Worker Options and then use the `CreateSession` API to create a Context object that can be passed to the calls to spawn Activity Executions.
```

File: sessions/workflow_test.go [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+type UnitTestSuite struct {
+	suite.Suite
+	testsuite.WorkflowTestSuite
+}
+
+func TestUnitTestSuite(t *testing.T) {
+	suite.Run(t, new(UnitTestSuite))
+}
+
+// Test_SomeFileProcessingWorkflow is a function to test the Workflow
```
Comment (djmagee) [inline sessions/workflow_test.go:22]:
```suggestion
// Test_SomeFileProcessingWorkflow is a function to test the Workflow.
```

File: sessions/README.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# sessions Temporal documentation Go SDK sample
+
+This application showcases the use of the Session APIs available in the Temporal Go SDK.
+
+The sample is contains information about usage code structure within the code comments.
+
+# How to run
+
+1. Make sure you have a Temporal Cluster running or Temporal Cloud to connect to.
+See the [Dev guide](https://docs.temporal.io/application-development/foundations#run-a-development-cluster) for the most up-to-date development option.
```
[the remark this answers is not in the document]
Comment (flossypurse) [reply sessions/README.md:10]: dprint / prettier formatting fixes this in docs site.

**Pull request merged**